popde/adaptive_kde.py

```diff
@@ -117,6 +116,88 @@ def set_adaptive_parameter(self, new_alpha, new_global_bw):
         self.set_alpha(new_alpha)
 
 
+class KDEOptimization(AdaptiveBwKDE):
+    """
+    Optimize bandwidth and alpha by grid search using
+    cross validation with a log likelihood figure of merit 
+    """
+    def __init__(self, data, bandwidth_options, alpha_options, weights=None, input_transf=None,
+                 stdize=False, rescale=None, backend='KDEpy', bandwidth=1.0, alpha=0.0,
+                 dim_names=None, do_fit=False, n_splits=2):
+        self.alpha_options = alpha_options
+        self.bandwidth_options = bandwidth_options
+        self.n_splits = n_splits
+        super().__init__(data, weights, input_transf, stdize,
+                         rescale, backend, bandwidth, alpha, dim_names)
+
+    def loo_cv_score(self, bandwidth_val, alpha_val):
+        from sklearn.model_selection import LeaveOneOut
+        loo = LeaveOneOut() 
+        fom = 0.
+        for train_index, test_index in loo.split(self.data):
+            train_data, test_data = self.data[train_index], self.data[test_index]
+            local_weights = None # FIX ME
+            awkde = AdaptiveBwKDE(train_data, local_weights, bandwidth=bandwidth_val,alpha=alpha_val)
```

Comment:
so here, although we have `input_transf=None, stdize=False, rescale=None` as defaults, it may be that the user initializes the class with some input transformations, and then we will want to apply these to the optimization as well (otherwise we will not get suitable bw/alpha values).  So please also pass on the values of `self.input_transf`, `self.stdize` and `self.rescale` here

Reply:
so I am bit confuse here. so I can add these for the data but when we evaluate kde on testing data (for score) do we need to also use evaluate_with_transf.

Reply:
Can you try to work it out for yourself please ...
We have the original data and we have the transformed data used for KDE fitting
So when doing the cross validation there are some choices that make sense and some that do not
Make one choice that makes sense and code it and I will review the code

Reply:
I did make changes and tested the code.